### PR TITLE
Refactor roles

### LINKED ISSUE #266: Towards authorization
two "small" changes:

1. rename `Group` to `Role`.
1. change `_userGroups :: [Group]` to `_userRole :: Role`.

### PR DESCRIPTION
**Pull request by fisx**
- rename Group to Role
- every user has always exactly one Role
- give local types Role, EditUser in Frontent.Page.Admin better names

Fixes #266

### CODE REVIEW

File: src/Types.hs [hunk 1]
```diff
@@ -633,15 +633,11 @@ isWild (IdeaLocationTopic _ _) = False
...
-groupLabel (ClassGuest _) = "Gast (Klasse)"
-groupLabel SchoolGuest    = "Gast (Klasse)"
-groupLabel Moderator      = "Moderator"
-groupLabel Principal      = "Direktor"
-groupLabel Admin          = "Administrator"
+-- | german role name
+roleLabel :: IsString s => Role -> s
+roleLabel (Student _)    = "Student"
+roleLabel (ClassGuest _) = "Gast (Klasse)"
+roleLabel SchoolGuest    = "Gast (Klasse)"
```
Comment (Mikolaj) [inline src/Types.hs]: Just checking: should names these really be equal? No "Schule" in the second one?

File: src/Frontend/Page/Admin.hs [hunk 2]
```diff
@@ -462,27 +461,13 @@ adminSettingsGaPUserEdit uid = redirectFormHandler editUserPage editUser
...
+payloadToUserRole :: EditUserPayload -> Role
+payloadToUserRole (EditUserPayload RoleStudent clss) = Student clss
+payloadToUserRole (EditUserPayload RoleGuest   clss) = ClassGuest clss
 
-    addGroup xs = case role of
-        RoleStudent -> Student clss : xs
-        RoleGuest   -> ClassGuest clss : xs
-
-isClassInGroup :: SchoolClass -> Group -> Bool
+isClassInGroup :: SchoolClass -> Role -> Bool
```
Comment (Mikolaj) [inline src/Frontend/Page/Admin.hs]: Shouldn't it renamed to `isClassInRole`?

**Comment by Mikolaj** [comment]
Other than the comments, LGTM.

File: src/Types.hs (hunk 1 above)
Comment (andorp) [inline src/Types.hs]: As far as I know the separation is made by us. The requirements don't do it, but for technical reasons it is more convenient to handle this way. The `SchoolGuest` is for the researchers, the other for the parents.

File: src/Frontend/Page/Admin.hs (hunk 2 above)
Comment (andorp) [inline src/Frontend/Page/Admin.hs]: Good catch

File: src/Types.hs (hunk 1 above)
Comment (fisx) [inline src/Types.hs]: and @Mikolaj you are right: that was a copy&paste error.  Will fix!

File: tests/Frontend/Page/AdminSpec.hs [hunk 3]
```diff
@@ -1,131 +0,0 @@
-module Frontend.Page.AdminSpec
```
Comment (andorp) [inline tests/Frontend/Page/AdminSpec.hs]: My little test suite. R.I.P :(

**Comment by andorp** [comment]
LGTM

**Pull request merged**